### PR TITLE
Enable setting of stream context options

### PR DESCRIPTION
**Pull request by slax0rr**
Some situations, namely mine, require to set some stream context options. This PR enables just that. User has to set an option array at initialization with an array that fits requirements of http://php.net/manual/en/function.stream-context-set-option.php

### CODE REVIEW

File: src/Mail/SmtpMailer.php [hunk 1]
```diff
@@ -41,6 +41,9 @@ class SmtpMailer extends Nette\Object implements IMailer
 	/** @var bool */
 	private $persistent;
 
+	/** @var array */
+	private $context;
```
Comment (JanTvrdik) [inline src/Mail/SmtpMailer.php]: should be probably named `$streamContext`

File: src/Mail/SmtpMailer.php [hunk 2]
```diff
@@ -58,6 +61,7 @@ public function __construct(array $options = array())
 		if (!$this->port) {
 			$this->port = $this->secure === 'ssl' ? 465 : 25;
 		}
+		$this->context = isset($options['context']) ? $options['context'] : array();
```
Comment (JanTvrdik) [inline src/Mail/SmtpMailer.php]: Maybe `$this->context = isset($options['context']) ? $options['context'] : NULL;`

File: src/Mail/SmtpMailer.php [hunk 3]
```diff
@@ -122,6 +126,12 @@ protected function connect()
 		if (!$this->connection) {
 			throw new SmtpException($error, $errno);
 		}
+
+		// set context options
+		if ($this->context !== array()) {
+				stream_context_set_option($this->connection, $this->context);
+		}
```
Comment (JanTvrdik) [inline src/Mail/SmtpMailer.php]: * the comment is useless,
* the condition should be just `if ($this->context) {`
* the indentation is wrong

**Comment by dg** [comment]
Please add a test.

File: src/Mail/SmtpMailer.php [hunk 4]
```diff
@@ -122,6 +126,11 @@ protected function connect()
 		if (!$this->connection) {
 			throw new SmtpException($error, $errno);
 		}
+
+		if ($this->streamContext) {
+			stream_context_set_option($this->connection, $this->context);
```
Comment (hrach) [inline src/Mail/SmtpMailer.php]: context vs. streamContext.

**Comment by slax0rr** [comment]
Oh god. This is embarrasing. I'm currently in a lot of work and really little time, I'll redo the whole PR tomorrow or next week.

**Comment by dg** [comment]
@slax0rr is it relevant after this https://github.com/nette/mail/commit/3b8ba96407029d94d83177086bcb5e1fe0046929?

**Comment by dg** [comment]
Closed by #30

**Pull request closed without merging**